src/jdk.compiler/share/classes/com/sun/tools/javac/jvm/Gen.java

```diff
@@ -912,7 +912,7 @@ public boolean isConstantDynamic(Symbol sym) {
     public void genArgs(List<JCExpression> trees, List<Type> pts) {
         for (List<JCExpression> l = trees; l.nonEmpty(); l = l.tail) {
             genExpr(l.head, pts.head).load();
-            if (pts.head.isNonNullable()) {
+            if (pts.head.isNonNullable() && l.head.type.constValue() == null) {
```

Comment:
These changes should become unnecessary if `isNonNullable` does the right thing.